packages/components/pie-toggle-switch/src/defs.ts

```diff
@@ -1,6 +1,15 @@
 import { RTLComponentProps } from '@justeattakeaway/pie-webc-core';
 
-export interface ToggleSwitchProps extends RTLComponentProps {
+export const labelPositions = ['leading', 'trailing'] as const;
+
+export type LabelPosition = typeof labelPositions[number];
+
+export type LabelProps = {
```

Comment:
I actually don't know if I'd prefer a label position prop separate from the label text. I'm really not a fan of the complexity added by these object prop definitions but I appreciate its a matter of preference